### PR TITLE
[Documentation] Concepts: Customer & ShopUser

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | yes
| New docs?    | no
| BC breaks?      | no
| Related tickets | partially #5275 
| License         | MIT

### CODE REVIEW

File: docs/book/customer_and_shopuser.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
+.. index::
+   single: Customer and ShopUser
+
+Customer and ShopUser
+=====================
+
+For handling customers of your system **Sylius** is using a combination of two entities - **Customer** and **ShopUser**.
+The difference between these two entities is simple:
+the **Customer** is unregistered and the **ShopUser** is registered in the system - they have an account.
```
Comment (lchrusciel) [inline docs/book/customer_and_shopuser.rst]: `the **Customer** is a guest in our shop and the **ShopUser** is a registered user in the system - he has an account.`

File: docs/book/customer_and_shopuser.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+``billingAddress`` and ``shippingAddress``, ``firstName``, ``lastName`` or ``birthday``.
+
+.. note::
+
+   The relation between the Customer and User is bidirectional. Both entities hold a reference to each other.
+
+ShopUser
+--------
+
+ShopUser entity is  designed for customers that have registered in the system - they have an account with both e-mail and password.
```
Comment (lchrusciel) [inline docs/book/customer_and_shopuser.rst]: `is  designed` -> `is designed`

File: docs/book/customer_and_shopuser.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+   $this->container->get('sylius.repository.shop_user')->add($user);
+
+Changing the ShopUser password
+''''''''''''''''''''''''''''''
+
+The already set password of a **ShopUser** can be easily changed via the ``setPlainPassword()`` method.
+
+.. code-block:: php
+
+   $user->getPlainPassword(); // returns 'pswd'
```
Comment (lchrusciel) [inline docs/book/customer_and_shopuser.rst]: It is not 100% true. AFAIR this method is valid only before persisting user to the database(or to the moment when a `eraseCredentials` function would be called). Usually, you are not able to get users plain password for security reasons. You are only allowed to get password via `getPassword` method, but this one will be encrypted.

File: docs/book/customer_and_shopuser.rst (hunk 1 above)
Comment (TheMadeleine) [inline docs/book/customer_and_shopuser.rst]: 👍  but - *they have an account ;>

Comment (lchrusciel) [inline docs/book/customer_and_shopuser.rst]: Sure, I forgot about our convention. _Mea culpa_

File: docs/book/customer_and_shopuser.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
+.. index::
+   single: Customer and ShopUser
+
+Customer and ShopUser
+=====================
+
+For handling customers of your system **Sylius** is using a combination of two entities - **Customer** and **ShopUser**.
+The difference between these two entities is simple:
+the **Customer** is a guest in your shop and the **ShopUser** is a registered user in the system - they have an account.
```
Comment (NeverResponse) [inline docs/book/customer_and_shopuser.rst]: Either `is a user registered in the system`, `is a registered user of the system` or just plain `is a registered user`, `is a registered user in the system` sounds funky.

File: docs/book/customer_and_shopuser.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+   $customer->setEmail('customer@test.com');
+
+   $this->container->get('sylius.repository.customer')->add($customer);
+
+The Customer entity can of course hold other information besides an email, it can be for instance
+``billingAddress`` and ``shippingAddress``, ``firstName``, ``lastName`` or ``birthday``.
+
+.. note::
+
+   The relation between the Customer and User is bidirectional. Both entities hold a reference to each other.
```
Comment (NeverResponse) [inline docs/book/customer_and_shopuser.rst]: 🚔  🚨  `User` => `ShopUser` 🚨  🚔

File: docs/book/customer_and_shopuser.rst [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+ShopUser entity is designed for customers that have registered in the system - they have an account with both e-mail and password.
+They can visit and modify their account.
+
+While creating new account the existence of the provided email in the system is checked - if the email was present - it will already have a Customer
+therefore the existing one will be assigned to the newly created ShopUser, if not - a new Customer will be created together with the ShopUser.
+
+How to create a ShopUser programmatically?
+''''''''''''''''''''''''''''''''''''''''''
+
+Assuming that you have a Customer (either retrieved from the reposiotry or a new one created) - use a factory to create
```
Comment (NeverResponse) [inline docs/book/customer_and_shopuser.rst]: `repository`;
`or newly created` perhaps?

File: docs/book/checkout.rst [hunk 7]
```diff
@@ -78,7 +78,7 @@ Firstly if the **Customer** is not yet set on the Order it will be assigned depe
 
 .. hint::
 
-    If you are do not understand how Users and Customers work in Sylius got to :doc:`Users Concept documentation </book/users_and_groups>`.
+    If you are do not understand how Users and Customers work in Sylius got to :doc:`Users Concept documentation </book/customer_and_shopuser>`.
```
Comment (NeverResponse) [inline docs/book/checkout.rst]: holla back that `are` 💃

File: docs/book/customer_and_shopuser.rst [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+=====================
+
+For handling customers of your system **Sylius** is using a combination of two entities - **Customer** and **ShopUser**.
+The difference between these two entities is simple:
+the **Customer** is a guest in your shop and the **ShopUser** is a user registered in the system - they have an account.
+
+Customer
+--------
+
+The Customer entity was created to collect data about non-registered guests of the system - one that has been buying without having an account
```
Comment (NeverResponse) [inline docs/book/customer_and_shopuser.rst]: `... - ones that have...`

Comment (NeverResponse) [inline docs/book/customer_and_shopuser.rst]: Besides this whole paragraph sounds pretty sinister, like we're some evil corporation which wants to acquire personal data of off everyone 😈

Comment (TheMadeleine) [inline docs/book/customer_and_shopuser.rst]: Just like every company :P

File: docs/book/checkout.rst (hunk 7 above)
Comment (lchrusciel) [inline docs/book/checkout.rst]: got? -> `go`

Comment (lchrusciel) [inline docs/book/checkout.rst]: Whole this sense doesn't seem to be right... We should mention, that this concept is related to the shop part only.

**Comment by pjedrzejewski** [comment]
Thank you Magda!

**Pull request merged**